pkg/webui/components/new-form/field/index.js

```diff
@@ -0,0 +1,210 @@
+
+import React from 'react'
+import bind from 'autobind-decorator'
+import classnames from 'classnames'
+import { getIn } from 'formik'
+
+import from from '../../../lib/from'
+import Icon from '../../icon'
+import Message from '../../../lib/components/message'
+import FormContext from '../context'
+import PropTypes from '../../../lib/prop-types'
+
+import style from './field.styl'
+
+export function getPassThroughProps (props, excludeProps) {
+  const rest = {}
+  for (const property of Object.keys(props)) {
+    if (!excludeProps[property]) {
+      rest[property] = props[property]
+    }
+  }
+  return rest
+}
+
+@bind
+class FormField extends React.Component {
+
+  static contextType = FormContext
+  static propTypes = {
+    className: PropTypes.string,
+    name: PropTypes.string.isRequired,
+    warning: PropTypes.message,
+    required: PropTypes.bool,
+    description: PropTypes.message,
+    onChange: PropTypes.func,
+    component: PropTypes.oneOfType([
+      PropTypes.func,
+      PropTypes.string,
+    ]).isRequired,
+  }
+
+  static defaultProps = {
+    onChange: () => null,
+    warning: '',
+    description: '',
+    required: false,
+  }
+
+  componentDidMount () {
+    const { name } = this.props
+
+    this.context.registerField(name, this)
+  }
+
+  componentWillUnmount () {
+    const { name } = this.props
+
+    this.context.unregisterField(name)
+  }
+
+  handleChange (value) {
+    const { name, onChange } = this.props
+    const { setFieldValue } = this.context
+
+    // check if the value is react's synthetic event
+    let newValue = value
+    if (typeof value === 'object' && 'target' in value) {
+      const target = value.target
+      if ('checked' in target) {
+        newValue = target.checked
+      } else if ('value' in target) {
+        newValue = target.value
+      }
+    }
+
+    setFieldValue(name, newValue)
+    onChange(value)
+  }
+
+  handleBlur (event) {
+    const { name } = this.props
+
+    this.context.handleBlur(event)
+    if (this.context.validateOnBlur) {
+      this.context.setFieldTouched(name, true)
+    }
+  }
+
+  render () {
+    const {
+      className,
+      name,
+      title,
+      warning,
+      description,
+      disabled,
+      required,
+      readOnly,
+      component: Component,
+    } = this.props
+    const { horizontal, disabled: formDisabled } = this.context
+
+    const fieldValue = getIn(this.context.values, name)
+    const fieldError = getIn(this.context.errors, name)
+    const fieldTouched = getIn(this.context.touched, name)
+    const fieldDisabled = disabled || formDisabled
+
+    const hasError = Boolean(fieldError)
+    const hasWarning = Boolean(warning)
+    const hasDescription = Boolean(description)
+
+    const showError = fieldTouched && hasError
+    const showWarning = !hasError && hasWarning
+    const showDescription = !showError && !showWarning && hasDescription
+
+    const fieldMessage = showError ? (
+      <div className={style.messages}>
```

Comment:
We could try to render this div by default with a fixed `min-height`, so that the validation messages will not result in a jump.